### PR TITLE
validate: Add non-empty checks 

### PR DESCRIPTION
**Pull request by q384566678**
Signed-off-by: zhouhao zhouhao@cn.fujitsu.com

### CODE REVIEW

File: validate/validate.go [hunk 1]
```diff
@@ -96,11 +96,15 @@ func NewValidatorFromPath(bundlePath string, hostSpecific bool, platform string)
 // CheckAll checks all parts of runtime bundle
 func (v *Validator) CheckAll() (msgs []string) {
 	msgs = append(msgs, v.CheckPlatform()...)
-	msgs = append(msgs, v.CheckRootfsPath()...)
+	if v.spec.Root != nil {
```
Comment (Mashimiao) [inline validate/validate.go]: This is not a perfect fix, should fix as in #428

Comment (q384566678) [inline validate/validate.go]: ok. I'll delet it , and please fix in #428 .

**Comment by Mashimiao** [comment]
LGTM

File: validate/validate.go [hunk 2]
```diff
@@ -100,7 +100,9 @@ func (v *Validator) CheckAll() (msgs []string) {
 	msgs = append(msgs, v.CheckMandatoryFields()...)
 	msgs = append(msgs, v.CheckSemVer()...)
 	msgs = append(msgs, v.CheckMounts()...)
-	msgs = append(msgs, v.CheckProcess()...)
+	if v.spec.Process != nil {
+		msgs = append(msgs, v.CheckProcess()...)
```
Comment (wking) [inline validate/validate.go]: `Hooks` is a similar pointer, and we're currently handling that by [unconditionally adding `CheckHooks`][1] and then [guarding against the `nil` inside `CheckHooks`][2].  I don't have a strong preference about whether the nil guard is inside `CheckAll` or the more specific `Check*` method, but I think we want to stay consistent.

[1]: https://github.com/opencontainers/runtime-tools/blob/2d270b8764c02228eeb13e36f076f5ce6f2e3591/validate/validate.go#L104
[2]: https://github.com/opencontainers/runtime-tools/blob/2d270b8764c02228eeb13e36f076f5ce6f2e3591/validate/validate.go#L175

Comment (Mashimiao) [inline validate/validate.go]: There are many places are not consistent, we can fix this in a follow PR.

Comment (liangchenye) [inline validate/validate.go]: @Mashimiao I think we can do it in this PR. Since the PR does not do anything else.
@q384566678 How about it?

Comment (q384566678) [inline validate/validate.go]: ok, I'll do it.

Comment (q384566678) [inline validate/validate.go]: updated.

**Comment by Mashimiao** [comment]
please squash them into one patch. there is no need to work with two patches

**Comment by liangchenye** [comment]
LGTM

**Comment by Mashimiao** [comment]
LGTM

**Pull request merged**